src/releases.ts

```diff
@@ -41,78 +44,109 @@ export async function uploadRelease(client: AxiosInstance, config: ReleaseConfig
     external_url: `${baseGateWayURL}/${releasePath}`,
     platforms: {},
   };
-  CliUx.ux.action.start('uploading files');
 
-  const platformsToSign: Partial<Record<SupportedPlatform, DesktopPlatform | WebPlatform>> = {};
+  CliUx.ux.action.start('Uploading files');
+
   for (const platformEntry of updatedPlatformEntries) {
     const platformKey = platformEntry.platform as SupportedPlatform;
-    const { path, executable } = platformEntry;
-    const file = fs.createReadStream(path);
+    const { path: platformPath, executable } = platformEntry;
+
+    // Handle WebGL folder upload, otherwise treat as a single file
+    const isWebGL = platformKey === 'webgl';
+    const files = isWebGL ? await getFolderFiles(platformPath) : await getSingleFile(platformPath);
```

Comment:
We should try to avoid adding if statements throughout the code for various platform key specific scenarios. It keeps us from making more general interfaces and the code is harder to read/more complicated

Could we add a field on the config hyperplay.yml file for `isFolderUpload` or something like that?

Then we would get its value here like 
```
const isFolderUpload = config.platforms[platformKey].isFolderUpload
const files = isFolderUpload ? await getFolderFiles(platformPath) : await getSingleFile(platformPath);
```

Reply:
I generally agree, though I still think it it's a balance between clean code and developer experience, realistically we only are going to be support a very small set of platforms for the forceable future, I think it's way more worth it to provide as minimal a configure and as good of a developer experience before purity of interfaces, or future state code complexity.

I think isFolderUpload is a great idea but if the intended purpose of the webgl builds is for it to upload a folder of webgl assets it seems like it only creates more friction to make the suer configure additional fields.

Either way I can it's a good to have config, just wanted to express some thought's around dev ux vs code purity.